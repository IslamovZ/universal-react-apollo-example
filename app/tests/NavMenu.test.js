import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/lib/test-utils';
import { print } from 'graphql-tag/printer';

import NavMenuWithData, { MENU_QUERY, NavMenu } from 'components/NavMenu/NavMenu';

describe('NavMenu Container', () => {
  it('renders without crashing', () => {
    const output = renderer.create(
      <MockedProvider>
        <NavMenuWithData />
      </MockedProvider>,
    );
    expect(output.toJSON()).toMatchSnapshot();
  });
});

describe('NavMenu Query', () => {
  it('should match expected shape', () => {
    expect(print(MENU_QUERY)).toMatchSnapshot();
  });
});

describe('NavMenu Component', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <NavMenu menu={[{ id: 1, title: 'ok' }]} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
