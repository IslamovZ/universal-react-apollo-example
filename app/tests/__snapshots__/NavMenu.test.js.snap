exports[`NavMenu Component renders correctly 1`] = `
<ul
  className={undefined}>
  <li>
    <a
      className={undefined}
      onClick={[Function]}>
      ok
    </a>
  </li>
</ul>
`;

exports[`NavMenu Container renders without crashing 1`] = `
<div>
  Loading....
</div>
`;

exports[`NavMenu Query should match expected shape 1`] = `
"query getMenu($lang: String!) {
  wp_query {
    menu(name: $lang) {
      id
      title
    }
  }
}
"
`;
